### PR TITLE
state sync: tune request timeout and chunkers

### PR DESCRIPTION
**Pull request by alexanderbez**
closes: #6564

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -827,8 +829,10 @@ func (cfg *StateSyncConfig) TrustHashBytes() []byte {
 // DefaultStateSyncConfig returns a default configuration for the state sync service
 func DefaultStateSyncConfig() *StateSyncConfig {
 	return &StateSyncConfig{
-		TrustPeriod:   168 * time.Hour,
-		DiscoveryTime: 15 * time.Second,
+		TrustPeriod:         168 * time.Hour,
+		DiscoveryTime:       15 * time.Second,
+		ChunkRequestTimeout: time.Minute,
+		ChunkFetchers:       1,
```
Comment (cmwaters) [inline config/config.go:835]: Are you sure we want to take these defaults. Having 1 chunk fetcher means that the process is no longer done in parallel which could really slow down state sync.

**Review by cmwaters — APPROVED** [review]
👍 

I introduced a similar concept with reverse sync as I modelled it off how chunks are sent and received so there's also 4 `blockFetchers` and a timeout of 10 seconds. Do you think there's any need to make these configurable.

File: config/config.go [hunk 2]
```diff
@@ -843,32 +847,47 @@ func (cfg *StateSyncConfig) ValidateBasic() error {
...
 		}
+
 		_, err := hex.DecodeString(cfg.TrustHash)
 		if err != nil {
 			return fmt.Errorf("invalid trusted-hash: %w", err)
 		}
+
+		if cfg.ChunkRequestTimeout == 0 {
+			return errors.New("chunk-request-timeout is required")
+		}
```
[the remark this answers is not in the document]
Comment (alexanderbez) [reply config/config.go:884]: Sure. What do you propose?

File: config/config.go (hunk 1 above)
Comment (alexanderbez) [reply config/config.go:835]: Yeah, good catch. We should keep the original values (10s/4).

File: config/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmwaters) [reply config/config.go:884]: I think either 5 or maybe 10 seconds should be enough

**Review by tychoish — APPROVED** [review]

**Comment by alexanderbez** [comment]
> 👍
> 
> I introduced a similar concept with reverse sync as I modelled it off how chunks are sent and received so there's also 4 `blockFetchers` and a timeout of 10 seconds. Do you think there's any need to make these configurable.

Yeah, I think so. If you feel it's reasonable 👍

**Comment by cmwaters** [comment]
We could use the same config i.e. amount of `ChunkFetchers` equals the amount of `LightBlockFetchers` and just call it `Fetchers` or something appropriate since I imagine we'll want them to be the same values

**Comment by cmwaters** [comment]
Just as a side thought, the priority of chunks is 1 (lowest possible). Would it make sense, if there are networks who want to have specific seed nodes for providing state that they are also able to change the priority here. I have a feeling - although not properly tested - that the occasional failures we see arise because the peer receives the request and then prepares the chunk but then the envelope gets dropped before being sent

**Pull request merged**